tests/Jobs/RockTheVote/ImportRockTheVoteRecordTest.php

```diff
@@ -70,4 +70,102 @@ public function testDoesNotCreateOrUpdatePostIfCompletedPostFound()
 
         ImportRockTheVoteRecord::dispatch($this->faker->rockTheVoteReportRow(), $this->faker->randomDigitNotNull);
     }
+
+    /**
+     * Test that user is updated if their voter registration status should change.
+     *
+     * @return void
+     */
+    public function testUpdatesUserIfShouldChangeStatus()
+    {
+        $userId = $this->faker->northstar_id;
+        $postId = $this->faker->randomDigitNotNull;
+
+        $this->mockGetNorthstarUser([
+            'id' => $userId,
+            'voter_registration_status' => 'uncertain',
+        ]);
+        $this->northstarMock->shouldNotReceive('createUser');
+        $this->rogueMock->shouldReceive('getPost')->andReturn([
+            'data' => [
+                0 => [
+                    'id' => $postId,
+                    'status' => 'uncertain',
+                ],
+            ],
+        ]);
+        $this->rogueMock->shouldNotReceive('createPost');
+        $this->northstarMock->shouldReceive('updateUser')->with($userId, [
+            'voter_registration_status' => 'registration_complete',
+        ]);
+        $this->rogueMock->shouldReceive('updatePost')->with($postId, [
+            'status' => 'register-OVR',
+        ]);
+
+        ImportRockTheVoteRecord::dispatch($this->faker->rockTheVoteReportRow([
+            'Status' => 'Complete',
+            'Finish with State' => 'Yes',
+        ]), $this->faker->randomDigitNotNull);
+    }
+
+    /**
+     * Test that user is not updated if their voter registration status should not change.
+     *
+     * @return void
+     */
+    public function testDoesNotUpdatesUserIfShouldNotChangeStatus()
+    {
+        $this->mockGetNorthstarUser([
+            'voter_registration_status' => 'registration_complete',
+        ]);
+        $this->northstarMock->shouldNotReceive('createUser');
+        $this->northstarMock->shouldNotReceive('updateUser');
+        $this->rogueMock->shouldReceive('getPost')->andReturn([
+            'data' => [
+                0 => [
+                    'id' => $this->faker->randomDigitNotNull,
+                    'status' => 'register-OVR',
+                ],
+            ],
+        ]);
+        $this->rogueMock->shouldNotReceive('createPost');
+        $this->rogueMock->shouldNotReceive('updatePost');
+
+        ImportRockTheVoteRecord::dispatch($this->faker->rockTheVoteReportRow([
+            'Status' => 'Step 1',
+            'Finish with State' => 'No',
+        ]), $this->faker->randomDigitNotNull);
+    }
+
+    /**
+     * Test that status should update when update value has higher priority than the current.
+     *
+     * @return void
+     */
+    public function testShouldUpdateStatus()
+    {
+        $statusHierarchy = [
+            'uncertain',
+            'ineligible',
+            'unregistered',
+            'confirmed',
+            'register-OVR',
+            'register-form',
+            'registration_complete',
+        ];
```

Comment:
Do you think this potentially indicates the (non-critical) need for this hierarchy to be a config or some formalized declaration somewhere?

Reply:
I like the idea moving this into the config! Can revisit in a future PR.